### ISSUE TITLE
Classes to indicate position of auto overlay

### ISSUE THREAD

**Issue by baleeds** [IgniteUI/igniteui-angular#9481]
Our designs need to change based on whether a panel has opened above or below the anchor point.

Here is what we have:
![image](https://user-images.githubusercontent.com/9682964/117309774-bc051b00-ae50-11eb-9fea-914666889d5a.png)

We need the top border and bottom border-radius to be swapped when the panel opens to the top like this.

Ideally, there could be a class to indicate that the panel is flipping up.

**Comment by wnvko** [comment]
There is no easy way to find out now when the direction was changed by the `AutoPositionStrategy`. IMO we should emit an event when this happens.

A little hacky way to handle this now could be to subscribe to overlay's `onAnimtaion` event and check there if position settings were changed. Check this Stackblitz where I've implemented this. Note this way uses internal properties and this may stop working if anything gets changed in the components.

Finally I will mark this as feature request.
